modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1450,6 +1450,26 @@ private void _validateName(
 
 			throw new ObjectFieldNameException.MustNotBeDuplicate(name);
 		}
+
+		List<ObjectRelationship> objectRelationships = ListUtil.concat(
+			_objectRelationshipPersistence.findByObjectDefinitionId1(
+				objectDefinition.getObjectDefinitionId()),
+			_objectRelationshipPersistence.findByObjectDefinitionId2(
+				objectDefinition.getObjectDefinitionId()));
+
+		for (ObjectRelationship objectRelationship : objectRelationships) {
+			if (!StringUtil.equals(name, objectRelationship.getName())) {
+				continue;
+			}
+
+			ObjectDefinition objectDefinition1 =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectRelationship.getObjectDefinitionId1());
+
+			throw new ObjectFieldNameException.
+				MustNotBeEqualsToExistingObjectRelationshipName(
+					objectDefinition1.getShortName());
```

Comment:
I believe that this error message, using the reference of the ObjectDefinition1 could sounds a bit confuse in some cases.
For example, if this name is equals to a relationship name which this ObjectDefinition be the child, the error message maybe sounds a bit nonsense. Because it will display to the user a shortName from an ObjectDefinition different from this one that he is managing now and isn't clear the link between them and what the user is doing wrong.